File: src/com/l2kt/gameserver/skills/effects/EffectSilentMove.java
package com.l2kt.gameserver.skills.effects;

import com.l2kt.gameserver.model.L2Effect;
import com.l2kt.gameserver.skills.Env;
import com.l2kt.gameserver.network.SystemMessageId;
import com.l2kt.gameserver.network.serverpackets.SystemMessage;
import com.l2kt.gameserver.templates.skills.L2EffectFlag;
import com.l2kt.gameserver.templates.skills.L2EffectType;
import com.l2kt.gameserver.templates.skills.L2SkillType;

public class EffectSilentMove extends L2Effect
{
	public EffectSilentMove(Env env, EffectTemplate template)
	{
		super(env, template);
	}
	
	@Override
	public boolean onStart()
	{
		super.onStart();
		return true;
	}
	
	@Override
	public void onExit()
	{
		super.onExit();
	}
	
	@Override
	public L2EffectType getEffectType()
	{
		return L2EffectType.SILENT_MOVE;
	}
	
	@Override
	public boolean onActionTime()
	{
		// Only cont skills shouldn't end
		if (getSkill().getSkillType() != L2SkillType.CONT)
			return false;
		
		if (getEffected().isDead())
			return false;
		
		double manaDam = calc();
		
		if (manaDam > getEffected().getCurrentMp())
		{
			SystemMessage sm = SystemMessage.getSystemMessage(SystemMessageId.SKILL_REMOVED_DUE_LACK_MP);
			getEffected().sendPacket(sm);
			return false;
		}
		
		getEffected().reduceCurrentMp(manaDam);
		return true;
	}
	
	@Override
	public int getEffectFlags()
	{
		return L2EffectFlag.SILENT_MOVE.getMask();
	}
}

File: src/com/l2kt/gameserver/skills/effects/EffectSilentMove.kt
package com.l2kt.gameserver.skills.effects

import com.l2kt.gameserver.model.L2Effect
import com.l2kt.gameserver.network.SystemMessageId
import com.l2kt.gameserver.network.serverpackets.SystemMessage
import com.l2kt.gameserver.skills.Env
import com.l2kt.gameserver.templates.skills.L2EffectFlag
import com.l2kt.gameserver.templates.skills.L2EffectType
import com.l2kt.gameserver.templates.skills.L2SkillType

class EffectSilentMove(env: Env, template: EffectTemplate) : L2Effect(env, template) {

    override fun onStart(): Boolean {
        super.onStart()
        return true
    }

    override fun onExit() {
        super.onExit()
    }

    override fun getEffectType(): L2EffectType {
        return L2EffectType.SILENT_MOVE
    }

    override fun onActionTime(): Boolean {
        if (skill.skillType !== L2SkillType.CONT)
            return false

        if (effected.isDead)
            return false

        val manaDam = calc()

        if (manaDam > effected.currentMp) {
            val sm = SystemMessage.getSystemMessage(SystemMessageId.SKILL_REMOVED_DUE_LACK_MP)
            effected.sendPacket(sm)
            return false
        }

        effected.reduceCurrentMp(manaDam)
        return true
    }

    override fun getEffectFlags(): Int {
        return L2EffectFlag.SILENT_MOVE.mask
    }
}